geomesa-accumulo/geomesa-accumulo-tools/src/main/scala/org/locationtech/geomesa/tools/accumulo/AccumuloRunner.scala

```diff
@@ -67,15 +68,25 @@ object AccumuloRunner extends Runner {
       }
     }
 
-    Option(command.params).collect { case p: AccumuloConnectionParams => p }.foreach { params =>
-      if (params.instance == null) {
-        params.instance = HdfsZooInstance.getInstance().getInstanceName
-      }
-      if (params.zookeepers == null) {
-        params.zookeepers = zookeepers
+    val params = Option(command.params)
+
+    params.collect {
+      case p: RequiredCredentialsParams if p.password == null => p
+      case p: OptionalCredentialsParams if p.password == null && p.user != null => p
+    }.foreach(_.password = Prompt.readPassword())
+
+    params.collect { case p: InstanceNameParams => p }.foreach { p =>
+      if (p.zookeepers == null) {
+        p.zookeepers = zookeepers
       }
-      if (params.password == null) {
-        params.password = Prompt.readPassword()
+      if (p.instance == null) {
+        p.instance = try {
+          // this will hang for 60+ seconds if it's not configured - so we wrap in a future and only wait 1s
+          import scala.concurrent.duration._
+          Await.result(Future(HdfsZooInstance.getInstance().getInstanceName)(ExecutionContext.global),  1000.millis)
+        } catch {
+          case NonFatal(e) => logger.warn(s"Exception getting zoo instance: ${e.toString}"); null
+        }
```

Comment:
either way - what in general are we back-porting? i was thinking just bug fixes, but I can open another PR for 1.2 if you want.